light_curves/code/heasarc_functions.py

```diff
@@ -1,12 +1,140 @@
 import pandas as pd
 from astroquery.heasarc import Heasarc
 from tqdm import tqdm
+from astropy.table import Table
+import pyvo
+import matplotlib.pyplot as plt
+import numpy as np
 
 from data_structures import MultiIndexDFObject
 
+#need to know the distribution of error radii for the catalogs of interest
+#this will inform the ligh curve query, as we are not intertsted in 
+#error radii which are 'too large' so we need a way of defining what that is
+def make_hist_error_radii(missioncat):
+    """plots a histogram of error radii from a HEASARC catalog
+    
+    Parameters
+    ----------
+    missioncat : str 
+        single catalog within HEASARC to grab error radii values  Must be one of the catalogs listed here: 
+            https://astroquery.readthedocs.io/en/latest/heasarc/heasarc.html#getting-list-of-available-missions
+    Returns
+    -------
+    heasarcresulttable : astropy table
+        results of the heasarc search including name, ra, dec, error_radius
+        
+    """
+    # get the pyvo HEASARC service.
+    heasarc_tap = pyvo.regsearch(servicetype='tap',keywords=['heasarc'])[0]
+
+    #simple query to select sources from that catalog
+    heasarcquery=f"""
+        SELECT TOP 5000 cat.name, cat.ra, cat.dec, cat.error_radius
+        FROM {missioncat} as cat
+         """
+    heasarcresult = heasarc_tap.service.run_sync(heasarcquery)
+
+    #  Convert the result to an Astropy Table
+    heasarcresulttable = heasarcresult.to_table()
+
+    #make a histogram
+    #zoom in on the range of interest
+    #error radii are in units of degrees
+    plt.hist(heasarcresulttable["error_radius"], bins = 30, range = [0, 10])
+    
+    #in case anyone wants to look further at the data
+    return heasarcresulttable
+    
+#example calling sequences
+#resulttable = make_hist_error_radii('FERMIGTRIG')
+#resulttable = make_hist_error_radii('SAXGRBMGRB')
```

Comment:
Can these go in the docstring for this function? I would put it between the first sentence and the parameters

Reply:
done